validation/chOpenApiValidator.go

```diff
@@ -0,0 +1,179 @@
+package validation
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/companieshouse/chs-delta-api/models"
+	"github.com/companieshouse/chs.go/log"
+	"github.com/getkin/kin-openapi/openapi3"
+	"github.com/getkin/kin-openapi/openapi3filter"
+	router "github.com/getkin/kin-openapi/routers/gorillamux"
+	"net/http"
+	"path/filepath"
+	"strings"
+)
+
+// Used for unit testing and mocking calls to external functions/methods.
+var (
+	callFilepathAbs                  = filepath.Abs
+	callNewRouter                    = router.NewRouter
+	callOpenApiFilterValidateRequest = openapi3filter.ValidateRequest
+	callFormatError                  = formatError
+)
+
+// CHValidator provides an interface to interact with the CH Validator.
+type CHValidator interface {
+	ValidateRequestAgainstOpenApiSpec(httpReq *http.Request, openApiSpec string) ([]byte, error)
+}
+
+// CHValidatorImpl is a concrete implementation of the CHValidator interface.
+type CHValidatorImpl struct {
+}
+
+// NewCHValidator returns a new CHValidator implementation.
+func NewCHValidator() CHValidator {
+	return CHValidatorImpl{}
+}
+
+// ValidateRequestAgainstOpenApiSpec takes a request and an openAPI spec location (string relative path) and uses the
+// spec to validate the provided request. If any validation errors are found, then they are formatted and returned to the
+// caller. If any errors are encountered while attempting to validate, they are handled and also returned to the caller.
+func (chv CHValidatorImpl) ValidateRequestAgainstOpenApiSpec(httpReq *http.Request, openApiSpec string) ([]byte, error) {
```

Comment:
Would it be possible for this validation package to be stored in the chs.go repository to enable re-use by other services?